### PR TITLE
manager user mixins full CRUD, read&update, and read

### LINKED ISSUE #85: create HasShopMixin
create **"HasShopMixin"**, this Mixin along with **is_stuff** will be added in any class next.

### PR DESCRIPTION
**Pull request by HMMK99**
without the Mixin
VirtualBox_Tomy_05_11_2020_09_21_20
with the Mixin
VirtualBox_Tomy_05_11_2020_09_20_10

### CODE REVIEW

File: xshop/users/mixins.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        if request.user.is_authenticated and (
+            request.user.type == [User.Types.MANAGER]
+            and len(Shop.objects.filter(mobile=request.user.mobile)) > 0
+        ):
+            return True
+
+    def has_view_permission(self, request, obj=None):
+        if request.user.is_authenticated and (
+            request.user.type == [User.Types.MANAGER]
+            and len(Shop.objects.filter(mobile=request.user.mobile)) > 0
```
Comment (shahwan42) [inline xshop/users/mixins.py]: `Shop.objects.filter(mobile=request.user.mobile).exists()` would be better

**Review by shahwan42 — CHANGES_REQUESTED** [review]

**Comment by shahwan42** [comment]
add extra validation in the User model to make sure that if the new user is `User.Types.Manager` must be  connected to a shop,

HELP
https://stackoverflow.com/questions/7366363/adding-custom-django-model-validation

**Review by shahwan42 — CHANGES_REQUESTED** [review]
could you please test it?

inside test_models in users.tests package
create a new ManagerTests class,
and test creating a manager without a shop raises a validation error inside a method

**Comment by shahwan42** [comment]
also, fix the failing tests in the CI

**Comment by HMMK99** [comment]
> could you please test it?
> 
> inside test_models in users.tests package
> create a new ManagerTests class,
> and test creating a manager without a shop raises a validation error inside a method

 I was working on it but the validation error keeps rising without testing

**Comment by shahwan42** [comment]
You need to edit code and add a shop to manager users inside the tests to satisfy the new requirement (shop connected to manager)

**Comment by HMMK99** [comment]
created with shop
VirtualBox_Tomy_10_11_2020_08_09_04
validation error without shop
VirtualBox_Tomy_10_11_2020_08_10_34

**Review by shahwan42 — APPROVED** [review]

**Pull request merged**